compiler/src/iree/compiler/Codegen/LLVMCPU/LLVMCPUTileAndFuseProducerConsumer.cpp

```diff
@@ -86,6 +86,14 @@ tileRootAndFuseProducerConsumer(IRRewriter &rewriter, TilingInterface rootOp,
 
   llvm::DenseSet<Operation *> yieldReplacementsFor;
   for (auto op : tiledAndFusedOps) {
+    // Fusion would clone the producer into the new loop op, which may lead to
+    // invalid dominance property.
+    bool isSameParent = llvm::all_of(op->getUsers(), [&](Operation *user) {
```

Comment:
This looks safe but I guess we could also check whether this would actually lead to a dominance issue to accommodate more cases? But that's maybe overkill/not needed for now?